json_serializable/lib/src/json_serializable_generator.dart

```diff
@@ -143,15 +146,16 @@ class JsonSerializableGenerator
         buffer.writeln('  ${field.type} get ${field.name};');
       }
 
-      var includeIfNull = annotation.read('includeIfNull').boolValue;
+      var classIncludeIfNull = annotation.read('includeIfNull').boolValue;
```

Comment:
I am not a huge fan of passing this around everywhere, it might make more sense to read the `JsonKey` annotations upfront and fill in defaults, so you can just pass a single configuration object around? Maybe a class that wraps up a `FieldElement` and a `JsonKey`?

Reply:
Done...a bit... At least for the class annotation